core/include/detray/navigation/navigator.hpp

```diff
@@ -495,12 +495,14 @@ class navigator {
         /// @returns next object that we want to reach (current target)
         DETRAY_HOST_DEVICE
         inline auto target() -> candidate_t & {
+            assert(static_cast<std::size_t>(m_next) < m_candidates.size());
             return m_candidates[static_cast<std::size_t>(m_next)];
```

Comment:
What is the type of `m_candidates`? Normally vector types have `at` function which runs the same assertion.

If so, you can just fix this line to m_candidates.at(static_cast<std::size_t>(m_next)) without any external assertions.

Reply:
It is a std::array and I rather don't want to run additional checking on device in release mode (vecmem device_vector also only does the bounds checking in debug mode)

Reply:
I am also not sure what happens if it wants to throws `std::out_of_range` exception on device?

Reply:
Why do you think it will run additional checking on device in release mode?

Reply:
I think this is what `at` does, or not?

Reply:
In `vecmem::device_vector` or `vecmem::device_array`, it only runs `assert`. I don't know about `std::array` but I guess it would just ignore the exception part. We can go with the separate assertion if we want to keep the performance for CPU release build